pet-api/pet-api/Infrastructure/DAL/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Dapper;
+using pet_api.Domain.Entities;
+using pet_api.Domain.Entities.Pagination;
+using pet_api.Domain.Interfaces;
+using System.Data;
+
```

Comment:
Need to add some error handling like try-catch

Reply:
I added an error handling into my UnitOfWork. Also errors are writed to the log file, it is very helpful feature